### PR TITLE
Feat: Feature Discovery Preview Data

### PR DESCRIPTION
**Pull request by danwom**
### Summary of Changes
Adds a new tab to the Feature page: Preview Data

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes.
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

**Comment by danwom** [comment]
Still need to add test coverage.

@devtagare @allisonsuarez @dkunitsk @Golodhros

File: frontend/amundsen_application/static/js/ducks/feature/reducer.ts [hunk 1]
```diff
@@ -86,6 +94,31 @@ export function getFeatureCodeFailure(
...
 }
 
+export function getFeaturePreviewData(
+  payload: FeaturePreviewQueryParams
+): GetFeaturePreviewDataRequest {
+  return {
+    payload,
+    type: GetFeaturePreviewData.REQUEST,
+  };
+}
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/ducks/feature/reducer.ts:104]: No tracking?

File: frontend/amundsen_application/static/js/features/PreviewData/index.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+}) => {
+  if (isLoading) {
+    return <PreviewDataLoader />;
+  }
+  if (
+    !previewData.columns ||
+    !previewData.data ||
+    previewData.columns.length === 0 ||
+    previewData.data.length === 0
+  ) {
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/features/PreviewData/index.tsx:62]: Nit: I would extract a variable named hasData from this logic and use it here

File: frontend/amundsen_application/static/js/features/PreviewData/index.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  return (
+    <div className="preview-data">
+      <div className="grid">
+        {previewData.columns.map((col, colId) => {
+          const fieldName = col.column_name;
+          return (
+            <div key={fieldName} id={fieldName} className="grid-column">
+              <div className="grid-cell grid-header subtitle-3">
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/features/PreviewData/index.tsx:77]: Please use the new classes for typography

File: frontend/amundsen_application/static/js/features/PreviewData/styles.scss [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    background-color: $body-bg-secondary;
+    height: $header-row-height;
+  }
+
+  .grid-cell {
+    align-items: center;
+    border-bottom: 1px solid $stroke;
+    display: flex;
+    flex-grow: 1;
+    font-size: 12px;
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/features/PreviewData/styles.scss:28]: can we use a variable here?

File: frontend/amundsen_application/static/js/features/PreviewData/styles.scss [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    height: $content-row-height;
+    justify-content: center;
+    min-width: 0;
+    padding: 8px;
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/features/PreviewData/styles.scss:32]: spacing-2

**Review by Golodhros — CHANGES_REQUESTED** [review]
Looking OK

I miss some tracking logic though

File: frontend/amundsen_application/static/js/ducks/feature/reducer.ts (hunk 1 above)
Comment (danwom) [reply frontend/amundsen_application/static/js/ducks/feature/reducer.ts:104]: The preview gets loaded automatically when the page loads. We have tracking on the tab component.

File: frontend/amundsen_application/static/js/features/PreviewData/index.tsx (hunk 2 above)
Comment (danwom) [reply frontend/amundsen_application/static/js/features/PreviewData/index.tsx:62]: Most of this is copy-pasted from the `DataPreviewButton`. I'll clean up a bit.

Comment (danwom) [reply frontend/amundsen_application/static/js/features/PreviewData/index.tsx:62]: TS compiler doesn't like it when I use a variable. It complains that a usage of `previewData.columns` below might be undefined.

Comment (Golodhros) [reply frontend/amundsen_application/static/js/features/PreviewData/index.tsx:62]: Ah... sometimes happens, TS!!

File: frontend/amundsen_application/static/js/stories/typography.story.tsx [hunk 6]
```diff
@@ -70,7 +70,7 @@ export const Typography = () => (
         <h1 className="title-3">Heading with .title-3</h1>
         <h1 className="subtitle-1">Heading with .subtitle-1</h1>
         <h1 className="subtitle-2">Heading with .subtitle-2</h1>
-        <h1 className="subtitle-3">Heading with .subtitle-3</h1>
+        <h1 className="text-subtitle-w3">Heading with .text-subtitle-w3</h1>
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/stories/typography.story.tsx:73]: Oh, this story is about the deprecated ones, there is another one with the new ones

**Review by Golodhros — APPROVED** [review]

File: frontend/amundsen_application/static/js/stories/typography.story.tsx (hunk 6 above)
Comment (danwom) [reply frontend/amundsen_application/static/js/stories/typography.story.tsx:73]: I see, I just did a global find and cleaned up the few usages. We should eventually cleanup all of the others.

**Pull request merged**